### DISCUSSION TITLE
Adding IfcOpenshell library to ChatGPT or other AIs

### DISCUSSION THREAD

**Discussion by Jordi-Rov** [IfcOpenShell/IfcOpenShell#3871] [Ideas]
Hi!
I have a question. I am trying to use ChaptGPT with IFC files and it tells me that it does not have the IfcOpenshell library installed. I have literally no idea about programming (that's why I try to use AI :-p) but, would it be interesting and useful to ask OpenAI to do it?

**Comment by Moult** [comment]
What exactly are you trying to do? I wasn't aware that ChatGPT executed any code.

**Reply by Jordi-Rov** [reply]
I use the "Advanced Data Analysis" functionality of GPT4 which, for example, executes python code to perform data analysis on files attached by the user. When attaching an IFC file the response I get is the following:
'In this specific environment provided by OpenAI, I do not have the ability to install additional packages or access external libraries that are not already included. Therefore, I cannot use ifcopenshell directly here.'
(again, apologies if I waste your time with questions that show my ignorance about programming)

**Reply by Jordi-Rov** [reply]
I have tried, for example, to analyze, view and modify GIS .gpkg files and it has been very useful

**Reply by aothms** [reply]
If I recall correctly GIS .gpkg are in fact SQLite Database file, for which python has built in support. Think of it as a schema on top of a well known file format. In theory IFC is the same. It's a schema that uses the ISO 10303 family of standards, but the difference is that usage of these are far from widespread, so no support in native python.

You might want to try using IfcJSON (using json), You can use BlenderBIM to convert your file. Or IfcOWL (using RDF, for which python does not have native support, but it's a lot more widely used). Try https://github.com/BenzclyZhang/IfcSTEP-to-IfcOWL-converters

In those ways maybe you can use the IFC schema using well known data exchange mechanisms.

Other than that I don't think we can provide extensive support for what you're trying to do.

**Reply by Jordi-Rov** [reply]
As far as I understand, the AI is now capable of reading the IFC file because it is in STEP format, so the problem is not reading the file but rather that it cannot use the functionalities offered by IfcOpenShell.

**Reply by Moult** [reply]
Unfortunately it sounds like can't give OpenAI access to additional packages. Your best bet is to transform the data into another serialisation.
